packages/lesswrong/server/editor/make_editable_callbacks.js

```diff
@@ -103,11 +103,42 @@ function wrapSpoilerTags(html) {
   return $.html()
 }
 
+const trimLeadingAndTrailingWhiteSpace = (html) => {
+  const $ = cheerio.load(`<div id="root">${html}</div>`)
+  const topLevelElements = $('#root').children().get()
+  // Iterate once forward until we find non-empty paragraph to trim leading empty paragraphs
+  removeLeadingEmptyParagraphsAndBreaks(topLevelElements, $)
+  // Then iterate backwards to trim trailing empty paragraphs
+  removeLeadingEmptyParagraphsAndBreaks(topLevelElements.reverse(), $)
+  return $("#root").html()
+}
+
+const removeLeadingEmptyParagraphsAndBreaks = (elements, $) => {
+   for (const elem of elements) {
+    if (isEmptyParagraphOrBreak(elem)) {
+      $(elem).remove()
+    } else {
+      break
+    }
+  }
+}
+
+const isEmptyParagraphOrBreak = (elem) => {
+  if (elem.name === "p") {
+    if (elem.children?.length === 0) return true
+    if (elem.children && elem.children[0]?.type === "text" && elem.children[0]?.data.trim() === "") return true
```

Comment:
Fails if the first child is a whitespace string, and there's more than one child. I'm not sure whether this can legally happen, but I can see it happening in cases like a paragraph that starts with leading spaces and then an <a> tag.

Reply:
```suggestion
    if (elem.children?.length === 1 && elem.children[0]?.type === "text" && elem.children[0]?.data.trim() === "") return true
```